src/pages/items-page/components/Pagination.jsx

```diff
@@ -0,0 +1,72 @@
+import styled from "@emotion/styled";
+import LeftArrowIcon from "@assets/icons/leftArrow";
+import RightArrowIcon from "@assets/icons/rightArrow";
+
+const Pagination = ({ totalCount, pageSize, page, onChange }) => {
+  const totalPage = Math.ceil(totalCount / pageSize);
+  const maxVisible = 5;
+  const half = Math.floor(maxVisible / 2);
+
+  let startPage = page - half;
+  let endPage = page + half;
+
+  if (startPage < 1) {
+    startPage = 1;
+    endPage = Math.min(totalPage, maxVisible);
+  }
+
+  if (endPage > totalPage) {
+    endPage = totalPage;
+    startPage = Math.max(1, totalPage - maxVisible + 1);
+  }
+
+  const tagList = [];
+  for (let i = startPage; i <= endPage; i++) {
+    tagList.push(
+      <PageItem key={i} onClick={() => onChange(i)} active={i === page}>
+        {i}
+      </PageItem>
+    );
+  }
```

Comment:
💊 제안
아래와 같이 작성해 가독성을 높일 수 있습니다~

```suggestion
  const tagList = Array.from(
    { length: endPage - startPage + 1 },
    (_, i) => {
      const pageNumber = startPage + i;
      return (
        <PageItem
          key={pageNumber}
          onClick={() => handlePageClick(pageNumber)}
          active={pageNumber === page}
        >
          {pageNumber}
        </PageItem>
      );
    }
  );
```

저는 개인적으로 jsx 문에서 바로 작성하시는 것이 가장 가독성에 좋을 것 같습니다.

```suggestion
 return (
  <PaginationList>
      <PageItem onClick={() => onChange(1)} /> 
          { Array.from({ length: endPage - startPage + 1 }).map(el => <PageItem .../>))}
      <PageItem onClick={() => onChange(totalPage)} />
  </PaginationList>
)
```